### ISSUE TITLE
Add support for `PULSE_*` environment variables

### ISSUE THREAD

**Issue by cipriancraciun** [lawl/NoiseTorch#31]
According to the Pulse FAQ and the source code, the server and client understands the following variables:

> https://www.freedesktop.org/wiki/Software/PulseAudio/FAQ/#index5h3
> https://github.com/pulseaudio/pulseaudio/blob/master/src/daemon/main.c#L248-L256

* `PULSE_SERVER` and `PULSE_COOKIE` -- used to connect to the pulse server;  (at the moment the `NoiseTorch` tool doesn't use these, and fails to find the running server;)
* `PULSE_CONFIG_PATH`, `PULSE_STATE_PATH`, `PULSE_RUNTIME_PATH` -- I assume `NoiseTorch` should use one of these (perhaps `PULSE_RUNTIME_PATH`) as it tries to access the `pid` file and it fails;  (I can't find the documentation now about these, but they are highlighted in the source code and do work;)

Alternatively add a support (either via custom environment variables or command line flags) for specifying where the Pulse "internals" exist, so that the user can properly point to them.

**Comment by cipriancraciun** [comment]
This issue could also be related to #23

**Comment by lawl** [comment]
Is there a distro that sets these?

I'd like to test this on a real system, not just me setting them how I think they work on my system.

**Comment by cipriancraciun** [comment]
I don't think there is a particular distribution setting these, however there are custom setups (especially among those using non mainstream desktop environments, or sessions started in non standard ways) that might have use for these.

(For example I run the `pulseaudio` as a service outside the X environment, on a volatile file-system, and thus I have to point it, and its clients, to the proper paths.)

(That is why I've said that perhaps the best approach would be to just expose via command line flags the "hardcoded" paths that `NoiseTorch` uses, as opposed to implementing the proper Pulse configuration "protocol".)

**Comment by lawl** [comment]
So, this is interesting, I looked at the source code for `pacmd` and while the docs say clients honor `PULSE_SERVER`, I cannot see it doing that in `pacmd`, nor can I see it doing any attempts at that when `strace`ing `pacmd` with `PULSE_SERVER` set.

So I think what we will do here is just following what `pacmd` does, which is search for a unix socket in this order: `PULSE_RUNTIME_PATH` -> `XDG_RUNTIME_DIR` -> (legacy) $HOME

**Comment by lawl** [comment]
I've implemented `PULSE_COOKIE` and `PULSE_RUNTIME_PATH`  you can find it either in [this branch](https://github.com/lawl/NoiseTorch/tree/respect-pulse-vars) or try this executable: 
[noisetorch.zip](https://github.com/lawl/NoiseTorch/files/4955033/noisetorch.zip)

Please let me know if this works for you.

**Comment by lawl** [comment]
Sorry, just found a bug, new binary: 
[noisetorch.zip](https://github.com/lawl/NoiseTorch/files/4955079/noisetorch.zip)

**Comment by lawl** [comment]
Any update?

**Comment by cipriancraciun** [comment]
@lawl thanks for the quick patches.  (I've seen the notifications, but didn't manage to test them out yesterday.)

I've just tried the attached file, and by just setting `PULSE_RUNTIME_PATH=...` and `PULSE_COOKIE=...` to the proper values works perfectly.

(In fact I've tried it with an empty environment, with only those two and `DISPLAY`, and the UI opened correctly.)

Thanks!  (I've used it today in a meeting and nobody complained.)  :)
